public/controllers/agent/components/register-agent.js

```diff
@@ -616,72 +914,402 @@ export const RegisterAgent = withErrorBoundary(
+            legend={legend}
+            options={options}
+            idSelected={idSelected}
+            onChange={onChange}
+            className={'osButtonsStyle'} />
+        )
+      }
+
+      const buttonGroupWithMessage = (legend, options, idSelected, onChange) => {
+        return (
+          <>
             <EuiButtonGroup
               color="primary"
-              legend="Choose the Operating system"
-              options={osButtons}
-              idSelected={this.state.selectedOS}
-              onChange={(os) => this.selectOS(os)}
+              legend={legend}
+              options={options}
+              idSelected={idSelected}
+              onChange={onChange}
+              className={'osButtonsStyle'}
             />
+            {this.state.selectedVersion == 'solaris10' || this.state.selectedVersion == 'solaris11' ? <EuiCallOut color="warning" className='message' iconType="iInCircle" title={
+              <span>
+                Might require some extra installation<EuiLink target="_blank" href="https://documentation.wazuh.com/current/installation-guide/wazuh-agent/wazuh-agent-package-solaris.html"> steps</EuiLink>.
+              </span>
+            }>
+            </EuiCallOut> : this.state.selectedVersion == '6.1 TL9' ? <EuiCallOut color="warning" className='message' iconType="iInCircle" title={
+              <span>
+                Might require some extra installation<EuiLink target="_blank" href="https://documentation.wazuh.com/current/installation-guide/wazuh-agent/wazuh-agent-package-aix.html"> steps</EuiLink>.
+              </span>
+            }>
+            </EuiCallOut> : this.state.selectedVersion == '11.31' ? <EuiCallOut color="warning" className='message' iconType="iInCircle" title={
+              <span>
+                Might require some extra installation<EuiLink target="_blank" href="https://documentation.wazuh.com/current/installation-guide/wazuh-agent/wazuh-agent-package-hpux.html"> steps</EuiLink>.
+              </span>
+            }>
+            </EuiCallOut> : <EuiCallOut color="warning" className='message' iconType="iInCircle" title={
+              <span>
+                The selected OS version reached its end of life (EOL). To install Wazuh follow our<EuiLink href="#"> guide</EuiLink>.
+              </span>
+            }>
+            </EuiCallOut>}
+          </>
+        )
+      }
+
+      const selectedVersionMac = (legend, options, idSelected, onChange) => {
+        return (
+          <EuiButtonGroup
+            color="primary"
+            legend={legend}
+            options={options}
+            idSelected={idSelected}
+            onChange={onChange}
+            className={'osButtonsStyleMac'} />
+        )
+      }
+
+      const steps = [
+        {
+          title: 'Choose the Operating system',
+          children: (
+            buttonGroup("Choose the Operating system", osButtons, this.state.selectedOS, (os) => this.selectOS(os))
```

Comment:
**nitpick**: Maybe we could use `operating system` in lowercase. I know this was already like this.

Reply:
Yes, it would be better, so I changed it there.